hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/IncrementalRelation.scala

```diff
@@ -82,9 +87,17 @@ class IncrementalRelation(val sqlContext: SQLContext,
 
   private val lastInstant = commitTimeline.lastInstant().get()
 
-  private val commitsTimelineToReturn = commitTimeline.findInstantsInRange(
-    optParams(DataSourceReadOptions.BEGIN_INSTANTTIME.key),
-    optParams.getOrElse(DataSourceReadOptions.END_INSTANTTIME.key(), lastInstant.getTimestamp))
+  private val commitsTimelineToReturn = {
+    if (useStateTransitionTime) {
```

Comment:
how does this affect the timetravel query? (which basically hits the same code path as inc query with same begin and end instant times)

Reply:
If I understand correctly, we only use begin and end instant time(which is `stateTransitionTime` if enabled) to filter instants, and still use instants' commit time to compare with `_hoodie_commit_time`

Reply:
Guess this could find out the out of order commits like compaction or clustering, but the assumption for `BEGIN_INSTANTTIME` and `END_INSTANTTIME` should be different, once `useStateTransitionTime` is true, user actually should uses the real transition time instead of the instant time to filter the instants.